source/server/server.cc

```diff
@@ -139,6 +145,7 @@ void InstanceImpl::flushStats() {
   server_stats_->days_until_first_cert_expiring_.set(
       sslContextManager().daysUntilFirstCertExpires());
 
+  stats_store_.mergeHistograms();
```

Comment:
If we do this, the merge will essentially always be one interval behind the flush interval. Is this what we want? Do we want to wait into the merge is complete? If so, my thinking was to have mergeHistograms() take a lambda which would fire when the merge is complete, and then the flush would complete.

Also, on this topic, for thought: I do wonder if the histogram merge interval should be separately configurable from the main stat flush interval. (E.g., let's say you wanted to flush stats/gauges every 500ms, you might want to wait longer to get more histogram data per interval). I would put in a TODO to consider this in the future.

Reply:
I did not get the first part. Since we block till all tls histograms are merged then main merge goes in sequentially - is the lambda still required?